### PR TITLE
Status update for Bitbucket server 

### PR DESCRIPTION
**Pull request by josmo**
Adding the ability for the bitbucket server remote to update the status of commits on the server

@bradrydzewski let me know if you have any concerns. I made it match bitbucket cloud as much as I could

### CODE REVIEW

File: remote/bitbucketserver/bitbucketserver.go [hunk 1]
```diff
@@ -149,8 +149,17 @@ func (c *Config) File(u *model.User, r *model.Repo, b *model.Build, f string) ([
 }
 
 // Status is not supported by the bitbucketserver driver.
-func (*Config) Status(*model.User, *model.Repo, *model.Build, string) error {
-	return nil
+func (c *Config) Status(u *model.User,r *model.Repo,b *model.Build,link string) error {
+	status := internal.BuildStatus{
+		State: convertStatus(b.Status),
+		Desc:  convertDesc(b.Status),
+		Key:   "Drone",
```
Comment (appleboy) [inline remote/bitbucketserver/bitbucketserver.go]: How about `fmt.Sprintf("Drone #%d - %s", b.Number, b.Branch)`?

Comment (appleboy) [inline remote/bitbucketserver/bitbucketserver.go]: screen shot 2016-08-14 at 3 27 54 pm

**Comment by josmo** [comment]
@appleboy check this one out and see if it works for you.  The name should update the label and we can keep the key the same :)

**Comment by appleboy** [comment]
@josmo LGTM !

**Pull request merged**